codegen/template.go

```diff
@@ -200,12 +201,21 @@ func (t *Template) GenerateClientFile(
 		IncludedPackages: m.IncludedPackages,
 		ClientID:         c.ClientID,
 	}
-	err := t.execTemplateAndFmt("http_client.tmpl", c.GoFileName, clientMeta)
```

Comment:
This is not being used and is instead used by the module system. 

As part of cleanup, can you delete this method and add the copyrightHeader to the module system ?